operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/managed/DefaultManagedDependentResourceContext.java

```diff
@@ -21,11 +21,11 @@ public <T> Optional<T> get(Object key, Class<T> expectedType) {
 
   @Override
   @SuppressWarnings("unchecked")
-  public <T> T put(Object key, T value) {
+  public <T> Optional<T> put(Object key, T value) {
     if (value == null) {
-      return (T) Optional.ofNullable(attributes.remove(key));
+      return (Optional<T>) Optional.ofNullable(attributes.remove(key));
```

Comment:
the cast and suppressions is ugly, if a mismatched type had been `put` in prior then the user would eventually get a classcast exception when the `Optional<String>` they were hoping for contains a different type.

I wonder if the type should become first class (part of the map key) so you can:

```
put("key", Long.class, 1L);
put("key", String.class, "hi");
get("key", String.class) -> Optional("hi");
get("key", Long.class) -> Optional(1L);
```
to sidestep type mismatch issues. But it might be a bit of a surprising API that it can have different values for a key.

Or we could return `Optional<?>` and put the typechecking work on the caller?